primitives/arithmetic/src/fixed128.rs

```diff
@@ -0,0 +1,675 @@
+// Copyright 2020 Parity Technologies (UK) Ltd.
+// This file is part of Substrate.
+
+// Substrate is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Substrate is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
+
+use codec::{Decode, Encode};
+use primitive_types::U256;
```

Comment:
Maybe we should investigate if we can avoid this or not. We have so far avoided including these primitive types in the default substrate runtime. We have our own minimal big-int primitives which should be enough for this case.

Reply:
how about feature flag it for now and refactor it later?

Reply:
Is switching to using the [`uint` crate](https://crates.io/crates/uint) and use `construct_uint!` directly? Seems like `U256` is used here just to handle overflows and the type is not exported in the public interface.